app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -257,6 +264,37 @@ class NavigationOverlayFragment : Fragment() {
         activity!!.moveTaskToBack(true)
     }
 
+    private fun observeAccountState(): Disposable {
+        val fxaRepo = serviceLocator.fxaRepo
+
+        return fxaRepo.accountState.subscribe { accountState ->
+            when (accountState) {
+                is AccountState.AuthenticatedWithProfile -> {
+                    val fxaProfile = accountState.profile
```

Comment:
nit: there's a lot of nesting here: maybe we can move the code inside this lambda into a helper function?